src/Microsoft.OpenApi/Models/OpenApiMediaType.cs

```diff
@@ -85,7 +98,7 @@ public void SerializeAsV3(IOpenApiWriter writer, OpenApiSpecVersion version = Op
             writer.WriteOptionalMap(OpenApiConstants.Encoding, Encoding, (w, e) => e.SerializeAsV3(w));
```

Comment:
the callback should be an argument as well.

Reply:
If I understand you correctly, according to the [spec](https://spec.openapis.org/oas/v3.0.1#media-type-object), the media type object doesn't include a callback object.

Reply:
That's not what I was referring to.
this `(w, e) => e.SerializeAsV3(w)` should be a parameter of the method instead of being hard coded is what I meant.

Reply:
I think this approach might make it a bit convoluted, the action delegate accepts different types as arguments when writing out different objects or mappings. I think it should stay as is.

Reply:
maybe but wouldn't keeping the hardcoded value lead to the implementer thinking they are supposed to serialize for v3 when we might be serializing for 3.1 for example?

Reply:
Oh, I see your point.